compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEUtils.cpp

```diff
@@ -85,6 +85,17 @@ FailureOr<std::array<uint32_t, 3>> getAIEIntegerMatmulInstructionSize(
   return it->second;
 }
 
+// The number of elements in a vector instruction for a given element type.
+FailureOr<uint32_t> getAIEMacWidth(Type inputType, Type outputType) {
+  // inputType = i8, outputType = i32:
+  if (inputType.isInteger(8) && outputType.isInteger(32)) return 32;
+  // inputType = i16, outputType = i32:
+  if (inputType.isInteger(16) && outputType.isInteger(32)) return 32;
+  // inputType = bf16, outputType = f32:
+  if (inputType.isBF16() && outputType.isF32()) return 16;
```

Comment:
Why the first two get value from bitwidth of the output and the last get it from the input?

Reply:
This was info I got from @jsetoain. Javier can you point us to the source of truth, please? Also can you please check if this API design makes sense longer term.

Reply:
> Why the first two get value from bitwidth of the output and the last get it from the input?

I believe that function is not returning the bitwidth of the vector element, it's returning the number of elements. For whatever design reason, the output of a mul/mac for `f32` is 512-bits, instead of the full 1024-bit width used by the integer unit. Both end up in the accumulator register bank, but the floating point unit produces half as many results per op as the integer unit[^1].

I suggest a different name might be less confusing. Instead of "MacWidth", maybe "MacNumElements" or "MacNumLanes"?

[^1]: Source of true: [AMD's AI Engine-ML Intrinsics User Guide; Multiply-accumulate of bfloat16 datatypes](https://www.xilinx.com/htmldocs/xilinx2023_2/aiengine_ml_intrinsics/intrinsics/group__intr__gpvectorop__mul__bf16xbf16.html)